pkg/systemd/logs.js

```diff
@@ -915,25 +913,94 @@ $(function() {
+            }
+        }
+    }
+
+    function parse_search(value) {
+        const new_items = {};
+        const values = value.split(" ")
+                .filter(item => {
+                    let s = item.split("=");
+                    if (s.length === 2 && s[0] === s[0].toUpperCase()) {
+                        new_items[s[0]] = s[1];
+                        return false;
+                    }
+
+                    const well_know_keys = ["start", "priority", "follow", "service", "identifier"];
+                    const map_keys = (key) => {
+                        if (key === "priority")
+                            return "prio";
+                        if (key === "identifier")
+                            return "tag";
+                        return key;
+                    };
+                    s = item.split(":");
+                    if (s.length === 2 && well_know_keys.includes(s[0])) {
+                        new_items[map_keys(s[0])] = s[1];
+                        return false;
+                    }
+
+                    return true;
+                });
+        new_items.grep = values.join(" ");
+        return new_items;
+    }
+
     $(cockpit).on("locationchanged", function() {
         update_services_list = true;
         update();
     });
 
-    $('#journal-current-day-menu a').on('click', function() {
-        update_services_list = true;
-        cockpit.location.go([], $.extend(cockpit.location.options, { start: $(this).attr("data-op") }));
-    });
-
     $('#journal-box').on('click', '.cockpit-logline', function() {
         var cursor = $(this).attr('data-cursor');
         if (cursor)
             cockpit.location.go([cursor], { parent_options: JSON.stringify(cockpit.location.options) });
     });
 
-    $('#journal-prio-menu a').on('click', function() {
+    $('#journal-current-day-menu').on('change', function() {
+        const options = parse_search(document.getElementById("journal-grep").value);
         update_services_list = true;
-        cockpit.location.go([], $.extend(cockpit.location.options, { prio: $(this).attr('data-prio') }));
+        cockpit.location.go([], $.extend(options, { start: $(this).val() }));
+    });
+
+    $('#journal-prio-menu').on('change', function() {
+        const options = parse_search(document.getElementById("journal-grep").value);
+        update_services_list = true;
+        cockpit.location.go([], $.extend(options, { prio: $(this).val() }));
+    });
+
+    $('#journal-service-menu').on("change", function() {
+        const options = parse_search(document.getElementById("journal-grep").value);
+        update_services_list = false;
+        cockpit.location.go([], $.extend(options, { tag: $(this).val() }));
+    });
+
+    $('#journal-follow').on("click", function() {
+        const state = $(this).attr("data-following") === "true";
+        if (state) {
+            $(this).text(_("Stream"));
```

Comment:
I demoed this to @garrett and @jgiardino today, and they suggested Resume might work better as a label than Stream, as it's a more common opposite verb to Pause, has clear analogues with playback in the physical world etc. Could even enforce that with the icons fa-play and fa-pause.
This is also what OpenShift does for their log streaming UI, so it's a gain for cross-product consistency.
I'll try to find a screenshot of that.

Reply:
Makes sense, updated!